### PR TITLE
add commit/apply duration for raft store

### PR DESCRIPTION
**Pull request by bufferflies**
Signed-off-by: bufferflies <1045931706@qq.com>

### What is changed and how it works?

Issue Number: Ref https://github.com/tikv/tikv/issues/12842

What's Changed:

```commit-message
```

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)
Apply duration
image

Apply wait duration
image

Side effects

### Release note 

```release-note
None.

```

### CODE REVIEW

File: components/raftstore-v2/src/fsm/apply.rs [hunk 1]
```diff
@@ -126,6 +128,7 @@ impl<EK: KvEngine, R: ApplyResReporter> ApplyFsm<EK, R> {
                 }
             }
             self.apply.flush();
+            STORE_APPLY_LOG_HISTOGRAM.observe(duration_to_sec(received_time.saturating_elapsed()));
```
Comment (BusyJay) [inline components/raftstore-v2/src/fsm/apply.rs:131]: This is not accurate as the fsm may be reschedule where there is an `await`.

File: components/raftstore-v2/src/operation/command/mod.rs [hunk 2]
```diff
@@ -78,6 +83,7 @@ pub struct CommittedEntries {
     /// Entries need to be applied. Note some entries may not be included for
     /// flow control.
     entry_and_proposals: Vec<(Entry, Vec<CmdResChannel>)>,
+    pub committed_time: Instant,
```
Comment (BusyJay) [inline components/raftstore-v2/src/operation/command/mod.rs:86]: Doesn't need to be `pub`.

**Review by BusyJay — APPROVED** [review]
Rest LGTM

File: components/raftstore-v2/src/fsm/apply.rs (hunk 1 above)
Comment (tonyxuqqi) [reply components/raftstore-v2/src/fsm/apply.rs:131]: From end-to-end perspective, I think it's the time the user request actually pays for the apply. So it's ok.

**Review by tonyxuqqi — APPROVED** [review]

File: components/raftstore-v2/src/operation/command/mod.rs (hunk 2 above)
Comment (bufferflies) [reply components/raftstore-v2/src/operation/command/mod.rs]: fixed

File: components/raftstore-v2/src/fsm/apply.rs (hunk 1 above)
Comment (BusyJay) [reply components/raftstore-v2/src/fsm/apply.rs:131]: No, it's not. It just the time apply finish a round of processing. Before the timer stops, user may have sent and received multiple round of req/resp.

Comment (bufferflies) [reply components/raftstore-v2/src/fsm/apply.rs:131]: Got it, I will using water fall to replace removing  this metrics in next pr.

**Pull request merged**